app/components/special-coverage/special-coverage-edit/special-coverage-edit-directive.js

```diff
@@ -66,8 +66,47 @@ angular.module('specialCoverage.edit.directive', [
           return promise;
         };
 
+        // Maps ID to Object for formatting (since only tunic_campaign_id stored on SpecialCoverage object)
+        $scope.tunicCampaignIdMapping = {};
+        $scope.campaignsRetrieved = function (response) {
+          response.data.results.forEach(function (result) {
+            $scope.tunicCampaignIdMapping[result.id] = result;
+          });
+          var tunicCampaignIds = [];
+          response.data.results.forEach(function (result) {
```

Comment:
Should consolidate and only run one of these `forEach` loops to take care of both actions